File: 백준/Silver/2583. 영역 구하기/영역 구하기.cc
#include <bits/stdc++.h>
using namespace std;
#define y1 aaaa
int m, n, k, x1,y1, x2,y2, a[101][101],visited[101][101];
int ret, area;
vector<int> areavec;

const int dy[4] = { -1, 0, 1, 0 };
const int dx[4] = { 0, 1, 0, -1 };

int dfs(int y, int x) {
	visited[y][x] = 1;
	area++;
	for (int i = 0; i < 4; i++) {
		int ny = y + dy[i];
		int nx = x + dx[i];
		if (ny < 0 || ny >= m || nx < 0 || nx >= n) continue;
		if (visited[ny][nx]) continue;
		if (a[ny][nx] == 1) continue;
		dfs(ny, nx);
	}
	return area;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	
	cin >> m >> n >> k;
	
	//초기화
	ret = 0;

	for (int i = 0; i < k; i++) {
		cin >> x1 >> y1 >> x2 >> y2;
		for (int y = y1; y < y2; y++) {
			for (int x = x1; x < x2; x++) {
				a[y][x] = 1;
			}
		}
	}
	for (int i = 0; i < m; i++) {
		for (int j = 0; j < n; j++) {
			if (visited[i][j]) continue;
			if (a[i][j] == 1) continue;
			area = 0;
			areavec.push_back(dfs(i, j));
			ret++;
		}
	}
	sort(areavec.begin(), areavec.end());
	cout << ret << '\n';
	for (int i : areavec) cout << i << " ";
	return 0;
}

File: 백준/Silver/2583. 영역 구하기/영역 구하기.js
const fs = require("fs")
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
let input = fs.readFileSync(filePath).toString().trim().split("\n")
const [m, n, k] =  input.shift().split(' ').map(Number)
// m, n, k
// y, x, k

const a = Array(101).fill().map(el => Array(101).fill(0));
const visited = Array(101).fill().map(el => Array(101).fill(0));
let cnt = 0;
let area = 0;
let ret = [];

const dy = [-1, 0, 1, 0];
const dx = [0, 1, 0, -1];

function dfs(y, x){
  area++;
  visited[y][x] = 1;
  for(let i=0; i<4; i++){
    const ny=y+dy[i];
    const nx=x+dx[i];
    if(ny<0 || ny>=m || nx<0 || nx>=n || visited[ny][nx]) continue;
    if(a[ny][nx] === 1) continue;
    dfs(ny, nx);
  }
}

for(let i=0; i<k; i++){
  const color = input.shift().split(' ').map(Number)
  for(let j=color[0]; j<color[2]; j++){
    for(let k=color[1]; k<color[3]; k++){
      a[k][j] = 1;
    }
  }
}

for(let i=0; i<m; i++){
  for(let j=0; j<n; j++){
    if(visited[i][j]) continue;
    if(a[i][j] === 1) continue;
    dfs(i, j);
    cnt++;
    ret.push(area)
    area=0;
  }
}
ret.sort((a, b) => a-b)
console.log(cnt);
console.log(ret.join(' '))